js/intro.js

```diff
@@ -0,0 +1,335 @@
+import { BASE_URL, HOME_URL } from './constants.js';
+
+const notAuthorized = document.querySelector('.not-authorized-page');
+const notFound = document.querySelector('.not-found-page');
+const mainContainer = document.querySelector('.intro-main');
+const loading = document.querySelector('.loading');
+const toastBox = document.querySelector('.toast-box');
+const queryString = window.location.search;
+const urlParams = new URLSearchParams(queryString);
+
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const res = {
+      data: await response.json(),
+      status: response.status,
+    };
+    return res;
+  } catch (err) {
+    throw err;
+  }
+}
+
```

Comment:
Can you move this function in utility file? so we can use this function in every place .